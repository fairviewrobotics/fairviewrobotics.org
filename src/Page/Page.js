import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import BackgroundImage from "../BackgroundImage/BackgroundImage";

import styles from './Page.module.css';

export default class Page extends PureComponent {

  static propTypes = {
    backgroundSrc: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired
  };

  render() {
    const {backgroundSrc, title, children} = this.props;
    return (
      <div>
        <BackgroundImage src={backgroundSrc}/>
        <h1 className={styles.title}>{title.toUpperCase()}</h1>
        {children}
      </div>
    );
  }
}
