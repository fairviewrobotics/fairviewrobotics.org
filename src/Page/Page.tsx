import React, { PureComponent } from 'react';
import BackgroundImage from "../BackgroundImage/BackgroundImage";
import styles from './Page.module.css';

export type PageProps = {
  backgroundSrc: string;
  title: string;
}

export default class Page extends PureComponent<PageProps> {
  render() {
    const {backgroundSrc, title, children} = this.props;
    return (
      <div>
        <BackgroundImage src={backgroundSrc}/>
        <h1 className={styles.title}>{title.toUpperCase()}</h1>
        {children}
      </div>
    );
  }
}
