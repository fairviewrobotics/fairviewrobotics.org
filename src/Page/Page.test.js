import React from "react";
import { shallow } from "enzyme";
import Page from "./Page";

describe('Page', () => {
  let wrapper;

  it('renders without crashing', () => {
    shallow(<Page backgroundSrc="backgroundimage.jpg" title="Cool Page!" />);
  });

  describe('given a page body, title, and background image', () => {
    beforeEach(() => {
      wrapper = shallow(
        <Page backgroundSrc="backgroundimage.jpg" title="Cool Page!">
          <div>
            <h1>Sub title of the page</h1>
            <p>With loads of useful robotics information</p>
          </div>
        </Page>
      );
    });

    it('should render the page', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});