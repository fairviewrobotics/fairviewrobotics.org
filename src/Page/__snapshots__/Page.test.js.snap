// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page given a page body, title, and background image should render the page 1`] = `
<div>
  <BackgroundImage
    fullScreen={false}
    src="backgroundimage.jpg"
  />
  <h1
    className="title"
  >
    COOL PAGE!
  </h1>
  <div>
    <h1>
      Sub title of the page
    </h1>
    <p>
      With loads of useful robotics information
    </p>
  </div>
</div>
`;
